[rertyy] iP

tzyian opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ]     Managing deadlines (coming soon)
- [ ]     Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tzyian committed:
level 0

tzyian committed:
Level-1

tzyian committed:
rename Duke to Numerator

tzyian committed:
Level-2

tzyian committed:
add task and mark, unmark

tzyian committed:
Level-4

tzyian committed:
update .gitignore

tzyian committed:
add scanner.hasNext() to aid input redirection

tzyian committed:
fix regex errors

tzyian committed:
add delete function

tzyian committed:
Move tasks to package

tzyian committed:
Update EXPECTED.TXT

tzyian committed:
Add Storage

tzyian committed:
Add back linebreaks between tasks

tzyian committed:
Merge branch 'branch-Level-7'

tzyian committed:
Add datetime parse functionality to Events and Deadlines

tzyian committed:
Update .gitignore

tzyian committed:
Merge remote-tracking branch 'origin/branch-Level-8'

tzyian committed:
Update .gitignore

tzyian committed:
Merge branch 'branch-Level-8'

tzyian committed:
Update README.md

tzyian committed:
Shift logic from Numerator.java to Parser.java

tzyian committed:
Add Command classes

tzyian committed:
Change ExitException to if-else checks

tzyian committed:
Merge branch 'A-MoreOOP'

tzyian committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tzyian committed:
Update build.gradle to use Numerator instead

tzyian committed:
Shift all files to package Numerator

tzyian committed:
Refactor files for class name coding style

tzyian committed:
Add unit tests for TaskList and ToDo

tzyian committed:
Merge branch 'A-JUnit'

tzyian committed:
Rename generated jar file name

tzyian committed:
Add findTasks method

tzyian committed:
Fix coding style violations

tzyian committed:
Add javadoc and header comments to public classes and variables

tzyian committed:
Merge branch 'branch-A-CodingStandard'

tzyian committed:
Fix merge conflicts

tzyian committed:
Fix merge conflicts

tzyian committed:
Extract methods from parseArguments

tzyian committed:
Add checkstyle dependencies

tzyian committed:
Fix most checkstyle violations

tzyian committed:
Extract lambda into new method

tzyian committed:
Change checkstyle version in build.gradle

tzyian committed:
Merge branch 'branch-A-CheckStyle'

tzyian committed:
Modify runtest.bat for new package structure

tzyian committed:
Simplify run logic

tzyian committed:
Add javafx dependencies

tzyian committed:
Set basic template for javafx

tzyian committed:
Fix name error

tzyian committed:
Add images

tzyian committed:
Update javadoc

tzyian committed:
Update boolean name to follow code style

tzyian committed:
Update skeleton code

tzyian committed:
Merge branch 'branch-Level-10'

tzyian committed:
Add title for javafx GUI

tzyian committed:
Removed redundant whitespace in a line

tzyian committed:
Add missing javadoc for getResponse method

The getResponse method helps to pass information from
the Numerator program to the UI. This method takes in
a String input and returns a String response to the
input.

Especially because this method will be used by other
packages, it is essential that the purpose of the
method is clear.

Adding javadoc helps to make the purpose clearer.

tzyian committed:
Add assertion to help validate filepath integrity

The load() attempts to read from the filepath if the
filepath points to a valid file. Otherwise, it
creates a new file at the filepath.

However, there may be existing regular files in the
filepath that may conflict with the creation of
the storage file. If the given filepath is invalid, the
application is unable to recursively modify the path
hierarchy to make the filepath valid.

Hence, the assertion is necessary to ensure filepath
validity.

tzyian committed:
Add assert statements to ensure method parameters are not null

tzyian committed:
Extract common code from Parser methods to separate method

tzyian committed:
Modify method name to be more informative

tzyian committed:
Modify method name to be more informative

tzyian committed:
Merge remote-tracking branch 'origin/branch-A-CodeQuality' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/numerator/Numerator.java

tzyian committed:
Fix merge conflict

tzyian committed:
Merge pull request #1 from rertyy/branch-A-Streams

Add Java Streams to code base

tzyian committed:
Merge pull request #2 from rertyy/branch-A-Assertions

Add assertions to code

tzyian committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/numerator/Numerator.java
#	src/main/java/numerator/Parser.java

tzyian committed:
Merge pull request #3 from rertyy/branch-A-CodeQuality

Increase code quality

tzyian committed:
Add initial Gradle workflow

laybach commented on src/main/java/numerator/Parser.java:
```diff
@@ -0,0 +1,202 @@
+
+    public static final String BYE_STRING = "Bye! Just close the window to exit";
+
+    private static final String INPUT_BYE = "bye";
+    private static final String INPUT_MARK = "mark";
+    private static final String INPUT_UNMARK = "unmark";
+    private static final String INPUT_DELETE = "delete";
+    private static final String INPUT_TODO = "todo";
+    private static final String INPUT_DEADLINE = "deadline";
+    private static final String INPUT_EVENT = "event";
+    private static final String INPUT_LIST = "list";
+    private static final String INPUT_FIND = "find";
```
You did great by starting the variable names that are similar with INPUT_!

laybach commented on src/main/java/numerator/Parser.java:
```diff
@@ -0,0 +1,202 @@
+     * @return the response to the user
+     * @throws NumeratorException if the input is not recognised
+     */
+    public static String parseArguments(String input, TaskList taskList, Storage storage) throws NumeratorException {
```
variable name for TaskList could be plural therefore you might consider using tasks instead of taskLIst

laybach commented on src/main/java/numerator/Parser.java:
```diff
@@ -0,0 +1,202 @@
+    }
+
+    private static String parseFind(String input, TaskList taskList) throws InputFormatException {
+        Matcher m = getMatcher(
```
variable name for Matcher can be longer to provide more details about what the variable is.

laybach commented on src/main/java/numerator/Storage.java:
```diff
@@ -0,0 +1,107 @@
+     * @return a TaskList object containing the tasks loaded from the file
+     * @throws LoadingException if there is an error loading the file
+     */
+    public TaskList load() throws LoadingException {
```
method name load is fine but can include more detail like loadFile()

laybach commented on src/main/java/numerator/Storage.java:
```diff
@@ -0,0 +1,107 @@
+     * @param taskList the TaskList object containing the tasks to be saved
+     * @throws SavingException if there is an error saving the file
+     */
+    public void save(TaskList taskList) throws SavingException {
```
same as the load method could specify more with saveFile()

laybach reviewed (commented):
LGTM! Just a few minor suggestion to take into considerations.

tzyian committed:
Change names load -> loadFile and save -> saveFile

tzyian committed:
Fix code quality violations

1. Extracted methods from code longer than 30 lines
2. Removed nesting of 3 layers or more

tzyian committed:
Fix checkstyle violation

tzyian committed:
Add tags

tzyian committed:
Fix checkstyle violation

tzyian committed:
Fix test cases from failing

tzyian committed:
Add Javadoc checkstyle via Regexp module

CS2103 checkstyle.xml does not check some of the CS2103 requirements for
 Javadoc format, hence necessitating additional checks.
This commit adds two checks under checkstyle.xml.

First one checks that the first word of every Javadoc must be
capitalised and end with an "s". Note: Triggers false positives for the
case for non-method Javadoc.

Second one checks that multiline Javadoc comments must end with a full
stop.

False positives and false negatives may occur.

CS2103 style guide can be found at
https://se-education.org/guides/conventions/java/intermediate.html

tzyian committed:
Fix XML errors

tzyian committed:
Merge branch 'checkstyle-custom'

tzyian committed:
Add checks for singleline Javadoc capital letters

tzyian committed:
Merge branch 'checkstyle-custom'

tzyian committed:
Fix Javadoc checkstyle violations

tzyian committed:
Fix loading error

tzyian committed:
Improve UI by increasing list max length

tzyian committed:
Add docs

tzyian committed:
Make small changes to GUI output

tzyian committed:
Update mainClass in build.gradle

tzyian committed:
Update README.md

tzyian committed:
Update README.md

tzyian committed:
Update build.gradle

tzyian committed:
Update .editorconfig
